packages/squiggle-lang/__tests__/compile/functions_test.ts

```diff
@@ -40,4 +40,15 @@ describe("Compile functions", () => {
 )`,
     { pretty: true, mode: "full" }
   );
+
+  // https://github.com/quantified-uncertainty/squiggle/issues/3141;
+  // check that annotations are referenced by stack, not captured
+  testCompile(
+    "foo = [1,5]; fn(x:foo) = x",
+    [
+      "(Assign foo (Array 1 5))",
+      "(Assign fn (Lambda (.parameters (.annotated x (StackRef 0))) (StackRef 0)))",
+    ],
+    { mode: "statements" }
+  );
```

Comment:
Seems kind of obvious, but I imagine it could be good to have some tests of just, 
"Here's a really complicated function that uses a bunch of techniques, that should get the right result".

I'm kind of surprised we didn't have tests that caught this before, makes me nervous about the other things we're not catching.

Reply:
There is already [a test](https://github.com/quantified-uncertainty/squiggle/blob/15f966689a16c75e85f93f5aee1109786a2c211e/packages/squiggle-lang/__tests__/compile/refs_test.ts) like this, for the most tricky part with nested captures, but it didn't have annotations.

Anything longer and the output becomes a black box that's hard to update or verify manually.